### DISCUSSION TITLE
resource on Android

### DISCUSSION THREAD

**Discussion by Pogzy** [beeware/toga#2772] [Q&A] [answered]
I am working on Windows but my goal is to create an apk for android, running the tutorial as an android worked fine, even on my real device.
On windows, I used a atlas.db file in resources folder and access it sucessfully using sqlite3 this way:
`con = sqlite3.connect(toga.App.app.paths.app / "resources/atlas.db")`
but after building for android, I've got a runtime error: 
> java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.astrofilter/org.beeware.android.MainActivity}: com.chaquo.python.PyException: OperationalError: unable to open database file

After unzipping the apk, my atlas.db file in not into the apk, do I have to declare personnal files somewhere for briefcase to find them ?
Is the real path file , and logical path toga.App.app.paths.app / "resources/atlas.db" on Windows and Android are the same? If not, how do I distinguish them at run time?

**Comment by freakboy3742** [comment, marked as answer]
You don't need to explicitly declare data files as part of packaging the app - any file that is a folder named in your sources (or a subfolder of those folders) will be copied into your app bundle.

The `toga.App.app.paths.app` location won't be *literally* the same path on Windows and Android because the app will be in different locations - but it will be *functionally* the same, relative to the location of the app.

If you're having difficulty reading a database file, that's likely because the code hasn't been updated and the database file hasn't been includes as part of the app. Re-running `briefcase update` (or `briefcase run -u`) should fix that problem.
